faceted_search/src/index.js

```diff
@@ -1,65 +1,168 @@
-	{label: "All text fields", field: "*", type: "text"},
-	{label: "Source", field: "source", type: "list-facet", facetSort:"index"},
-	{label: "Context", field: "hasContextCategory", type: "list-facet", facetSort:"count"},
-	{label: "Material", field: "hasMaterialCategory", type: "list-facet", facetSort:"count"},
-	{label: "Specimen", field: "hasSpecimenCategory", type: "list-facet", facetSort:"count"},
-	{label: "Registrant", field: "registrant", type: "list-facet", facetSort:"count"},
+    {label: "All text fields", field: "searchText", type: "text"},
+    {label: "Identifier", field: "id", type: "text"},
+    {label: "Source", field: "source", type: "list-facet", facetSort:"index"},
+    {label: "Context", field: "hasContextCategory", type: "list-facet", facetSort:"count"},
+    {label: "Material", field: "hasMaterialCategory", type: "list-facet", facetSort:"count"},
+    {label: "Specimen", field: "hasSpecimenCategory", type: "list-facet", facetSort:"count"},
+    {label: "Registrant", field: "registrant", type: "list-facet", facetSort:"count"},
 ];
 
 // The sortable fields you want
 const sortFields = [
-	{label: "Name", field: "koppelnaam_s"},
-	{label: "Date of birth", field: "birthDate_i"},
-	{label: "Date of death", field: "deathDate_i"}
+    {label: "Identifier", field: "id"},
+    {label: "Registrant", field: "registrant"},
 ];
 
 
 // Construct the solr client api class
-const solrClient = 	new SolrClient({
-	url: config.solr_url,
-	searchFields: fields,
-	sortFields: sortFields,
+const solrClient = new SolrClient({
+    url: config.solr_url,
+    searchFields: fields,
+    sortFields: sortFields,
+    rows: 20,
+    pageStrategy: "paginate",
+    start: 0,
 
-	// Delegate change callback to redux dispatcher
-	onChange: (state) => store.dispatch({type: "SET_SOLR_STATE", state: state})
+    // Delegate change callback to redux dispatcher
+    onChange: (state) => {store.dispatch({type: "SET_SOLR_STATE", state: state})}
 });
 
+
+
+function APP() {
+	// https://reactrouterdotcom.fly.dev/docs/en/v6/api#usesearchparams
+	// Used for modifying the query string
+    let [searchParams, setSearchParams] = useSearchParams();
+
+	// https://reactjs.org/docs/hooks-state.html
+	// Simple boolean to remember whether we've already decoded state or not (not sure if this will work in all cases, but continue investigating)
+    // let [curState, setCurState] = useState(false);
+
+
+	// A note about when this gets called -- https://reactjs.org/docs/hooks-reference.html#useeffect
+	// This is called asynchronously after a render is complete.  The rule is that render itself must be
+	// stateless, so state mutating operations need to be contained here.  In our case the sequence is
+	// (1) User clicks something in the UI (like enables or disables a search facet )
+	// (2) UI refreshes
+	// …some time passes
+	// (3) This gets called, and we write back the current query string to the browser's location using setSearchParams
+    useEffect(() => {
+
```

Comment:
Please get rid of the whole ``` For learning ``` part of the code.  It's checked into git and we can recover it if we ever need it.

Reply:
Get it.